### ISSUE TITLE
Flow restarts and crashes react-native application without calling any error callback 

### ISSUE THREAD

**Issue by diogob** [onfido/onfido-android-sdk#104]
#### What was the expected behaviour?

The flow should proceed to the next step

#### What happened instead?

The SDK restart the flow from the first image capture screen. Also, the react application seems to die without any logging and without calling any error callback.

#### Version info:

- `onfido-capture-sdk` version: 5.5.0
- Android API version(s):

- `compileSdkVersion`: 28
- `targetSdkVersion`: 28
- `minSdkVersion`: 16
- Support library version (if applicable): androidXCore = "1.0.2"

#### Steps to reproduce:

1. Grab the SDK token using applicant_id and application_id (using `applicationId` from `android/app/build.gradle`.

#### Onfido Flow initialization code:

The code bellow is used to create a `NativeModule` for react:
```java
package com.coinberry_mobile.OnfidoSDK;

import android.app.Activity;
import android.content.Intent;

import com.facebook.react.bridge.ActivityEventListener;
import com.facebook.react.bridge.BaseActivityEventListener;
import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.Callback;
import com.onfido.android.sdk.capture.Onfido;
import com.onfido.android.sdk.capture.ExitCode;
import com.onfido.android.sdk.capture.OnfidoConfig;
import com.onfido.android.sdk.capture.OnfidoFactory;
import com.onfido.android.sdk.capture.errors.OnfidoException;
import com.onfido.android.sdk.capture.upload.Captures;
import com.onfido.android.sdk.capture.ui.options.FlowStep;
import android.widget.Toast;

import androidx.annotation.NonNull;

import java.util.GregorianCalendar;

public class OnfidoSDK extends ReactContextBaseJavaModule {

    private static final String E_ACTIVITY_DOES_NOT_EXIST = "E_ACTIVITY_DOES_NOT_EXIST";
    private static final String E_FAILED_TO_SHOW_ONFIDO = "E_FAILED_TO_SHOW_ONFIDO";
    private final Onfido client;
    private final ReactApplicationContext context;
    private Callback mSuccessCallback;
    private Callback mErrorCallback;

    private final ActivityEventListener mActivityEventListener = new BaseActivityEventListener() {
        @Override
        public void onActivityResult(int requestCode, int resultCode, Intent data) {
            super.onActivityResult(requestCode, resultCode, data);

            client.handleActivityResult(resultCode, data, new Onfido.OnfidoResultListener() {
                @Override
                public void userExited(@NonNull ExitCode exitCode) {
                    if (mErrorCallback != null) mErrorCallback.invoke(exitCode.toString());
                }

                @Override
                public void userCompleted(@NonNull Captures captures) {
                    if (mSuccessCallback != null) mSuccessCallback.invoke(captures);
                }

                @Override
                public void onError(OnfidoException e) {
                    e.printStackTrace();
                    if (mErrorCallback != null) mErrorCallback.invoke(e.getMessage());
                }
            });
        }
    };

    public OnfidoSDK(ReactApplicationContext reactContext) {
        super(reactContext);
        client = OnfidoFactory.create(reactContext).getClient();
        reactContext.addActivityEventListener(mActivityEventListener);
        this.context = reactContext;
    }

    @Override
    public String getName() {
        return "OnfidoSDK";
    }

    @ReactMethod
    public void startSDK(String token, Callback successCallback, Callback errorCallback) {
        Activity currentActivity = getCurrentActivity();
        final FlowStep[] defaultSteps = new FlowStep[]{
                FlowStep.CAPTURE_DOCUMENT,
                FlowStep.CAPTURE_FACE                  //Face Capture Step
        };

        mSuccessCallback = successCallback;
        mErrorCallback = errorCallback;

        if (currentActivity == null) {
            if (mErrorCallback != null) mErrorCallback.invoke(E_ACTIVITY_DOES_NOT_EXIST);
            return;
        }

        try {
            OnfidoConfig onfidoConfig = OnfidoConfig.builder(this.context)
                    .withCustomFlow(defaultSteps)
                    .withSDKToken(token)
                    .build();

            client.startActivityForResult(currentActivity, 1, onfidoConfig);
        }
        catch (Exception e) {
            if (mErrorCallback != null) mErrorCallback.invoke(E_FAILED_TO_SHOW_ONFIDO);
            mErrorCallback = null;
        }
    }
}
```

On the react side I call that using typescript:
```typescript
interface OnfidoSDKParams {
    token: string;
}

interface ServerResponse {
    data: OnfidoSDKParams;
}

const getOnfidoToken = async (jwt): Promise<OnfidoSDKParams | null> => {
    try {
        const headers = {
            'Accept': 'application/json',
            'Authorization': `Bearer ${jwt}`,
            'Content-Type': 'application/json',
        };
        
        const applicationId = (Platform.OS === 'android' ? "com.coinberry_mobile" : "com.coinberry.coinberry");
        const response = await client.post<OnfidoSDKParams>("/onfido/tokens", { application_id: applicationId },
            { 
                headers,
                transformResponse: (r: ServerResponse) => r.data
            });
            console.log("Data: ", response.data);
            return response.data;
    } catch (error) {
        console.log("Error ", error);
        return null;
    }
};

        const jwt = await this.user.getToken();
        const credentials = await getOnfidoToken(jwt);
        if(credentials !== null)
        {
            NativeModules.OnfidoSDK.startSDK(
                credentials?.token,
                () => { console.log("Verification complete", "To perform another verification, press \"Launch\"") },
                (errorCause) => {
                    if (errorCause == "USER_LEFT_ACTIVITY") {
                        console.log("Flow cancelled", "To try again, press \"Launch\"")
                    } else {
                        console.log("Flow not finished", "To try again, press \"Launch\"")
                    }
                }
            );    
        } else {
            console.log("Error starting SDK");
        }

```

#### Issue severity information (number of users/crashes if applicable):

Crashes on every single attempt

#### Other useful info:

JWT Token used:
 ```
eyJhbGciOiJIUzI1NiJ9.eyJwYXlsb2FkIjoickQ2ODNoYmVRT1VqdXEycnpPYnpnV0J2WEc4dmNHb3Zic2g1K1ZNaEJSU2RzTm5FcUpGdHVDOVducEVCXG41cGpnVDFqMFBYTkpFSnQ4a2tCanZVK3JKNTJqTFp2c2xpRXYwbCt4S1BhZ3o3L1FScG1ybHVsU1l1YjFcbkx5WEVCbWJ1XG4iLCJ1dWlkIjoiNXpKc3M5dFk1NTUiLCJleHAiOjE1ODM1NDYxNjcsInVybHMiOnsib25maWRvX2FwaV91cmwiOiJodHRwczovL2FwaS51cy5vbmZpZG8uY29tIiwidGVsZXBob255X3VybCI6Imh0dHBzOi8vdGVsZXBob255LnVzLm9uZmlkby5jb20iLCJkZXRlY3RfZG9jdW1lbnRfdXJsIjoiaHR0cHM6Ly9zZGsudXMub25maWRvLmNvbSIsInN5bmNfdXJsIjoiaHR0cHM6Ly9zeW5jLm9uZmlkby5jb20iLCJob3N0ZWRfc2RrX3VybCI6Imh0dHBzOi8vaWQub25maWRvLmNvbSJ9fQ.3J4lNLz5Oo7GzjsvucfEaVDZmk0AbWK9d3rcncnsuzE
```

[bugreport-Nexus_5X_API_28_x86-2020-03-06-19-25-43-ee8354c8-b46c-430a-a3af-3781ea712170.zip](https://github.com/onfido/onfido-android-sdk/files/4301176/bugreport-Nexus_5X_API_28_x86-2020-03-06-19-25-43-ee8354c8-b46c-430a-a3af-3781ea712170.zip)

**Comment by diogob** [comment]
Sorry, I have just updated the above report since the `onfido-capture-sdk` version was incorrect. Now I report the version we actually tested `5.5.0`.
I also tried with `onfido-capture-sdk-core` to the same effect.

**Comment by gokhanaliccii** [comment]
@diogob  I've tried to reproduce the problem using our [sample app](https://github.com/onfido/onfido-android-sdk/tree/master/sample-app)  but I've completed the flow successfully. 

Could you please try to reproduce the problem with our [sample app](https://github.com/onfido/onfido-android-sdk/tree/master/sample-app)  and if you could share logcat output I'd appreciate

**Comment by diogob** [comment]
@gokhanaliccii thanks for the help. Looking at logcat output and using some knowledge from https://github.com/onfido/onfido-sdk-react-native-sample-app I was able to figure out the issue.

It turns out I was getting a `java.lang.NoClassDefFoundError: Failed resolution of: Lokhttp3/internal/Platform;` in a call from some fetcher class inside the onfido sdk.

I solved the issue adding the lines to `app/build.graddle`:
```
    implementation 'com.squareup.okhttp3:logging-interceptor:3.12.10'
    implementation 'com.squareup.okhttp3:okhttp:3.12.10'
```
